### PR TITLE
Fix IndexNotFound errors during inserts on partitioned table

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

testDeletePartitionWhileInsertingData in
PartitionedTableConcurrentIntegrationTest was flaky:

    io.crate.exceptions.RelationUnknown: Relation 'yfjvblyelpwzy..partitioned.parted.04132' unknown
            ...
    Caused by: [yfjvblyelpwzy..partitioned.parted.04132/d-U30GdTQN-GukNosYvp8A] org.elasticsearch.index.IndexNotFoundException: no such index
            at app//org.elasticsearch.cluster.routing.RoutingTable.shardRoutingTable(RoutingTable.java:150)
            at app//io.crate.planner.operators.InsertFromValues.execute(InsertFromValues.java:647)
            at app//io.crate.planner.operators.InsertFromValues.lambda$execute$0(InsertFromValues.java:285)
            at java.base@17/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
            at java.base@17/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2309)
            at app//io.crate.planner.operators.InsertFromValues.execute(InsertFromValues.java:281)
            at app//io.crate.action.sql.Session.singleExec(Session.java:664)
            at app//io.crate.action.sql.Session.exec(Session.java:527)
            at app//io.crate.action.sql.Session.triggerDeferredExecutions(Session.java:514)
            at app//io.crate.action.sql.Session.sync(Session.java:498)
            at app//io.crate.testing.SQLTransportExecutor.execute(SQLTransportExecutor.java:258)
            at app//io.crate.testing.SQLTransportExecutor.execute(SQLTransportExecutor.java:236)
            ... 5 more

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/io/crate/planner/operators/InsertFromValues.java [hunk 1]
```diff
@@ -642,11 +643,42 @@ private static ShardLocation getShardLocation(String indexName,
...
+                    .currentNodeId();
+            } catch (IndexNotFoundException e) {
+                lastFailure.set(e);
+                if (!IndexParts.isPartitioned(request.index())) {
+                    synchronized (compressedResult) {
+                        compressedResult.markAsFailed(request.items());
+                    }
+                }
+                countdown.accept(request);
+                continue;
```
Comment (seut) [inline server/src/main/java/io/crate/planner/operators/InsertFromValues.java:680]: just curious: why is the continue here needed?

**Review by seut — APPROVED** [review]
👍

File: server/src/main/java/io/crate/planner/operators/InsertFromValues.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/planner/operators/InsertFromValues.java:680]: It should continue the loop, to have the same behavior as if the error happened in the actionListener.

Comment (seut) [reply server/src/main/java/io/crate/planner/operators/InsertFromValues.java:680]: I may phrased it confusing: why is the continue needed to continue the loop as this is afaik the last statement inside the loop?

Comment (mfussenegger) [reply server/src/main/java/io/crate/planner/operators/InsertFromValues.java:680]: Maybe the diff looks confusing? It's the first statement and the remainder of the loop body cannot work without a `nodeId`.

Comment (seut) [reply server/src/main/java/io/crate/planner/operators/InsertFromValues.java:680]: ahh right, my fault! thx for explaining :)

**Pull request merged**